node/subsystem-types/src/messages.rs

```diff
@@ -328,18 +328,13 @@ pub enum NetworkBridgeRxMessage {
 	NewGossipTopology {
 		/// The session info this gossip topology is concerned with.
 		session: SessionIndex,
-		/// Ids of our neighbors in the X dimensions of the new gossip topology,
-		/// along with their validator indices within the session.
-		///
-		/// We're not necessarily connected to all of them, but we should
-		/// try to be.
-		our_neighbors_x: HashMap<AuthorityDiscoveryId, ValidatorIndex>,
-		/// Ids of our neighbors in the X dimensions of the new gossip topology,
-		/// along with their validator indices within the session.
-		///
-		/// We're not necessarily connected to all of them, but we should
-		/// try to be.
-		our_neighbors_y: HashMap<AuthorityDiscoveryId, ValidatorIndex>,
+		/// Our validator index in the session, if any.
+		local_index: Option<ValidatorIndex>,
```

Comment:
Worth noting here, that this `ValidatorIndex` may extend beyond parachain validators as in  `TopologyPeerInfo`.
I still think the lack of type-safety here might bite us in the future.

Reply:
`SessionInfo` has `active_validator_indices: Vec<ValidatorIndex>` which is rarely used in node-side code, and it also has `validators: Vec<ValidatorId>`. `ValidatorIndex` is not a type-safe index but it exclusively is used to refer to the latter, not the former. And here it is guaranteed to not be out-of-bounds of `validators`.

Reply:
> And here it is guaranteed to not be out-of-bounds of validators

Is it though? I thought it's an index into discovery keys, which includes all active validators, not just parachain `validators`.

Reply:
Anyway, this could be addressed in https://github.com/paritytech/polkadot/issues/2403.